src/utils/parsers.rs

```diff
@@ -0,0 +1,87 @@
+pub use teloxide_macros::TelegramCommandEnum;
+/// Enum for telegram commands
+///
+/// Example:
+/// ```
+/// use teloxide::utils::TelegramCommandEnum;
+/// use teloxide::utils::parse_command_into_enum;
+/// #[derive(TelegramCommandEnum, PartialEq, Debug)]
+/// enum TelegramCommand {
+///     Start,
+///     Help,
+/// }
+/// let (command, args) = parse_command_into_enum::<TelegramCommand>("/", "/start arg1 arg2").unwrap();
+/// assert_eq!(command, TelegramCommand::Start);
+/// assert_eq!(args, vec!["arg1", "arg2"]);
+/// ```
+pub trait TelegramCommandEnum: Sized {
+    fn try_from(s: &str) -> Option<Self>;
+}
+
+pub fn parse_command_into_enum<'a, T>(
+    prefix: &str,
+    text: &'a str,
+) -> Option<(T, Vec<&'a str>)>
+where
+    T: TelegramCommandEnum,
+{
+    let (command, args) = parse_command(prefix,text)?;
+    return match T::try_from(command) {
+        Some(command) => Some((command, args)),
+        _ => None,
+    };
+}
+
+/// Function which parse string and return command with args
+///
+/// Example:
+/// ```
+/// use teloxide::utils::parse_command;
+/// let text = "/ban 5 hours";
+/// let (command, args) = parse_command("/", text).unwrap();
+/// assert_eq!(command, "ban");
+/// assert_eq!(args, vec!["5", "hours"]);
+/// ```
+pub fn parse_command<'a>(prefix: &str, text: &'a str) -> Option<(&'a str, Vec<&'a str>)> {
```

Comment:
I think we need two functions: `parse_command(text: &str)` and `parse_command_with_prefix(prefix: &str, text: &str)`. The first one is the simplification of `parse_command_with_prefix("/", text)`.

Reply:
Who need `prefix` other than `/`?

Reply:
fixed